### ISSUE TITLE
[V3 Say] Missing permissions badly handled

### ISSUE THREAD

**Issue by retke** [retke/Laggrons-Dumb-Cogs#31]
## Description

When using the `[p]say` command without any channel argument in a channel where the bot don't have the `send_messages` permission, it will try to tell the user it doesn't have enough permissions to send a message, in the same forbidden channel.

A possible fix to this would be to send a private message to the user, if the bot can't send the message in the context channel. If DMing the user is impossible, let the error propagate, hoping an error manager have been set up.

## Sentry logging

https://sentry.io/retke/say/issues/687992134/
```
Forbidden: FORBIDDEN (status code: 403): Missing Permissions
  File "say/say.py", line 165, in say
    await channel.send(text, files=files)
  File "discord/abc.py", line 756, in send
    embed=embed, nonce=nonce)
  File "discord/http.py", line 209, in request
    raise Forbidden(r, data)

Forbidden: FORBIDDEN (status code: 403): Missing Permissions
(1 additional frame(s) were not displayed)
...
  File "say/say.py", line 197, in _say
    await self.say(ctx, text, files)
  File "say/say.py", line 169, in say
    _("I am not allowed to send messages in ") + channel.mention, delete_after=1
  File "redbot/core/commands/context.py", line 57, in send
    return await super().send(content=content, **kwargs)
  File "discord/abc.py", line 761, in send
    data = await state.http.send_message(channel.id, content, tts=tts, embed=embed, nonce=nonce)
  File "discord/http.py", line 209, in request
    raise Forbidden(r, data)
```

**Comment by retke** [comment]
Probably gonna do this for 1.4.6, with a few Sentry changes, bringing more informations, and more suggestions if needed.

**Comment by retke** [comment]
Fixed with #33
